### PR TITLE
 Show empty chart and summary numbers when search returns no results 

### LINKED ISSUE #1451: Table search: Don't try to fill the report if we know there are no results
When doing a search with the table input, a query might return no results. In that case, we want to avoid sending API calls that we know will return no results or 404.

This issue consists on three parts:
- [x] Table #1452.
- [ ] Chart #1491.
- [ ] Summary numbers #1491.

### PR DESCRIPTION
**Pull request by Aljullu**
Fixes #1451.

Follow-up of #1452.

Show empty chart and summary numbers when a search returns no results and avoid doing useless API calls.

### Accessibility

- [x] I've tested using only a keyboard (no mouse)
- [x] I've tested using a screen reader

### Screenshots
image

### Detailed test instructions:
* Go to the _Products_ report.
* Make a search that will return no results.
* Verify the chart and summary numbers are empty.
* In your browser devtools _Network_ tab, verify there is no call sent to `/wc/v4/reports/products/stats`.

### CODE REVIEW

File: client/analytics/components/report-summary/index.js
Comment (joshuatf) [inline client/analytics/components/report-summary/index.js]: What do you think about adding this as a type directly to `formatValue()` [here](https://github.com/woocommerce/wc-admin/blob/847e9788f4f8581de6016dce39e023efe5d6bd93/packages/number/src/index.js#L49)?

Not sure if there are issues if we do this with interdependencies between packages since we'd need to import `formatCurrency()`.

File: client/analytics/components/report-summary/index.js
Comment (joshuatf) [inline client/analytics/components/report-summary/index.js]: Nice. Much more readable here.

File: client/analytics/components/report-summary/index.js
Comment (joshuatf) [inline client/analytics/components/report-summary/index.js]: Nitpicking here, but could we rename this to `emptyResults` or ``emptySearchResults`?  Someone else glancing at this might get confused and think this prop means no search query was made.

**Review by joshuatf — COMMENTED** [review]
Awesome work @Aljullu.  Only the search result query is made on this page now.

Left a comment about possibly moving the currency formatting into the numbers package and another super nitpicky comment about naming, but this is testing well!

File: client/analytics/components/report-summary/index.js
Comment (Aljullu) [reply client/analytics/components/report-summary/index.js]: Yes, that was my main concern as well, I didn't want to add `@woocommerce/currency` as a dependency in `@woocommerce/number`.

Looking around, I found that was discussed [here](https://github.com/woocommerce/wc-admin/pull/1399/commits/ce4729f3853a0bccd61085e2f72990b5e82bf2c4#r251197977) and the decision was not to add `type=currency` in `formatValue`. So I would leave it as it is now. How does it sound?

File: client/analytics/components/report-summary/index.js
Comment (Aljullu) [reply client/analytics/components/report-summary/index.js]: Good point, makes a lot of sense!

Fixed in 21c1490d47028d06a15a101c401c89bb21fcdf52.

**Comment by Aljullu** [comment]
@joshuatf thanks for the review! Feel free to take another look when you have a second!

_(Travis build is failing because of #1498)_

File: client/analytics/components/report-summary/index.js
Comment (joshuatf) [reply client/analytics/components/report-summary/index.js]: Sweet- thanks!

File: client/analytics/components/report-summary/index.js
Comment (joshuatf) [reply client/analytics/components/report-summary/index.js]: Good find on that discussion!  Ya, you definitely made the right call by keeping it separate.

**Review by joshuatf — APPROVED** [review]
Thanks for making that change and for posting the circular dependency discussion link!  This one looks good to merge after a rebase (#1498 is approved now).

**Pull request merged**
